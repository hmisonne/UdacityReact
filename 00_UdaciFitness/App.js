import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import AddEntry from './components/AddEntry'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import reducer from './reducers'

export default class App extends React.Component {
  
  render(){
     return (
      <Provider store={createStore(reducer)}>
        <View style={styles.container}>
          <AddEntry/>
        </View>
      </Provider>
  );
  }
 
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});