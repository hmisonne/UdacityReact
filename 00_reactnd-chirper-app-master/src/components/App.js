import React, { Component, Fragment } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { handleInitialData } from '../actions/shared'
import { connect } from 'react-redux'
import Dashboard from './Dashboard'
import NewTweet from './NewTweet'
import LoadingBar from 'react-redux-loading'
import TweetPage from './TweetPage'
import UpdateTweet from './UpdateTweet'
import Nav from './Nav'

class App extends Component {
	
	componentDidMount(){
		this.props.dispatch(handleInitialData())
	}
	render() {
		return (
			<Router>
				<Fragment>
					<LoadingBar />
					  <div className='container'>
					  
					  <Nav />
					  {this.props.loading === true 
					  	? null
					  	: 
					  	<div>
					  		<Route exact path='/' component={Dashboard}/>
					  		<Route path='/tweet/:id' component={TweetPage}/>
					  		<Route path='/new' component={NewTweet}/>
				  			<Route path='/update/:id' component={UpdateTweet}/>

					  	</div>
					  	
						   }
					  </div>
				
		  </Router>
		)
	}
}

function mapStateToProps ({authedUser}) {
	return {
		loading: authedUser === null
	}
}

export default connect(mapStateToProps)(App)

// <TweetPage match={{params: {id: '8xf0y6ziyjabvozdd253nd'}}}/>}

