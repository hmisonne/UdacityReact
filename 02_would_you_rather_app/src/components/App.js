import React, { Component, Fragment} from 'react';
import LoadingBar from 'react-redux-loading'
import { handleInitialData } from '../actions/shared'
import { setAuthedUser } from '../actions/authedUser'
import { connect } from 'react-redux'
import Dashboard from './Dashboard'
import Leaderboard from './Leaderboard'
import NewQuestions from './NewQuestions'
import QuestionDetails from './QuestionDetails'
import Nav from './Nav'
import Login from './Login'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import PropTypes from 'prop-types'


class App extends Component {
	static propTypes = {
	    dispatch: PropTypes.func.isRequired,
	    loading: PropTypes.bool.isRequired,
	  }
	componentDidMount(){
		this.props.dispatch(handleInitialData())
	}
	logout = () => {
		this.props.dispatch(setAuthedUser(null))
	}
	render(){
		const {authedUserName, users} =this.props
		if (authedUserName === null){
			return(
				<Login/>
				)
		}
		return (
			<Router>
				<Fragment>
					<LoadingBar />
					    <div className="container">
					    	<Nav 
					    		authedUserName={authedUserName}
					    		logout={this.logout}/>
					    	{this.props.loading === true 
							  	? null
							  	: 
							  	<div>
									<Route exact path='/' component={Dashboard}/>
									<Route exact path='/login' component={Login}/>
									<Route path='/add' component={NewQuestions}/>
									<Route path='/questions/:id' component={QuestionDetails}/>
							        <Route exact path='/leaderboard' component={Leaderboard}/>
							        
							    </div>
						}
				    </div>
			    </Fragment>
			</Router>
		  );
	}
  
}

function mapStateToProps ({ authedUser, users }) {
	// Retrieve authed username for NavBar
	let authedUserName = null
	if (users[authedUser] !== undefined) { authedUserName=users[authedUser].name }
	

	return {
		loading: authedUser === null,
		authedUserName,
	}
}

export default connect(mapStateToProps)(App);