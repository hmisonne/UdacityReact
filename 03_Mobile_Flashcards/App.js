import 'react-native-gesture-handler';
import * as React from 'react';
import { Platform, StatusBar, StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import * as Font from 'expo-font';
import { Ionicons } from '@expo/vector-icons';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import DeckView from './screens/DeckView'
import Quizz from './screens/Quizz'
import NewQuestion from './screens/NewQuestion'

import BottomTabNavigator from './navigation/BottomTabNavigator';

const Stack = createStackNavigator();


export default function App(props) {
    return (
      <View style={styles.container}>
        {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
        <NavigationContainer >
          <Stack.Navigator initialRouteName="Root">
            <Stack.Screen name="Root" component={BottomTabNavigator} />
            <Stack.Screen name="DeckView" component={DeckView} />
            <Stack.Screen name="Quizz" component={Quizz} />
            <Stack.Screen name="NewQuestion" component={NewQuestion} />
          </Stack.Navigator>
        </NavigationContainer>
      </View>
    );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
