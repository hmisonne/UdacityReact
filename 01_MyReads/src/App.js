import React from 'react'
import * as BooksAPI from './BooksAPI'
import './App.css'
import BookSelection from './BookSelection'
import SearchBooks from './SearchBooks'
import { Route } from 'react-router-dom'

const bookshelves = [
  {
    id: 1,
    name:"currentlyReading",
    formatedName:'Currently Reading'
  },
  {
    id: 2,
    name:"wantToRead",
    formatedName:'Want to Read'
  },
  {
    id:3,
    name:"read",
    formatedName:'Read'
  }]


class BooksApp extends React.Component {
  state = {
    books:[],
    BookIDtoShelf:{}
  }

  componentDidMount() {
    this.loadBooks()
  }

  loadBooks = () => {
    BooksAPI.getAll()
      .then((books) => {
        this.setState((currState)=> ({
          books
        }))
      })
      .then(() =>
        this.loadShelves()
        )
  }

  loadShelves = () => {
      this.state.books.map(book => this.updateBookIDtoShelf(book))
    }

  updateBookIDtoShelf = (book) => {
      this.setState(prevState => ({
        BookIDtoShelf: {...prevState.BookIDtoShelf,
          [book.id]: book.shelf}
      }))
  }

  updateBookShelfLocation = (book, shelf) => {
    BooksAPI.update(book, shelf)
      .then((bookIDperShelf) => 
        this.loadBooks());
    this.updateBookIDtoShelf(book)
  }


  render() {
    const {books, BookIDtoShelf} = this.state
    console.log('BookIDtoShelf App state',BookIDtoShelf)
    return (
      <div className="app">
        <Route exact path='/' render={()=> (
          <BookSelection
            bookshelves={bookshelves}
            bookSelection={books}
            updateBookShelfLocation={this.updateBookShelfLocation}
            BookIDtoShelf={BookIDtoShelf}
          />
          )} />
        <Route path='/search' render={() => (
          <SearchBooks 
            bookshelves={bookshelves}
            updateBookShelfLocation={this.updateBookShelfLocation}
            BookIDtoShelf={BookIDtoShelf}/>
          )}/>
      </div>
    )
  }
}

export default BooksApp


        