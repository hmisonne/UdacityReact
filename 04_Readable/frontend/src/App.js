import React, { Component, Fragment } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import './App.css';
import Home from './components/Home'
import CreateEdit from './components/CreateEdit'
import PostDetail from './components/PostDetail'
import Nav from './components/Nav'

class App extends Component {

    render() {
        return (
            <Router>
              <Fragment>
                <div className='container'>
                  <Nav />
                  <div className="App">
                    <Route exact path='/' component={Home}/>
                    <Route exact path='/:category' component={Home}/>
                    <Route exact path='/post/new/create' component={CreateEdit}/>
                    <Route exact path='/:category/:id' component={PostDetail}/>
                    <Route exact path='/:category/:id/update' component={CreateEdit}/>
                  </div>
                </div>
              </Fragment>
            </Router>
        );
    }
}

export default connect()(App)