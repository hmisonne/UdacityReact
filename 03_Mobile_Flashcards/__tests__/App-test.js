import * as React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

jest.mock('../navigation/MainNav', () => 'MainNav');

describe('App', () => {

  it(`renders the main screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
