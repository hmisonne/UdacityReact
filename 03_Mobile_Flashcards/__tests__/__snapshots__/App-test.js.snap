// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the main screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#6b7d70",
        "height": 44,
      }
    }
  />
  <MainNav />
</View>
`;
